src/honey_badger/epoch_state.rs

```diff
@@ -265,13 +257,13 @@ where
     /// epoch, moves those contributions into a batch, outputs the batch and updates the epoch.
     pub fn try_output_batch(&self) -> Option<(Batch<C, N>, FaultLog<N>)> {
         let proposer_ids = self.subset.accepted_ids()?;
-        let plaintexts: BTreeMap<N, &[u8]> = self
-            .decryption
-            .iter()
-            .flat_map(|(id, dec_state)| dec_state.plaintext().map(|pt| (id.clone(), pt)))
-            .collect();
-        if !proposer_ids.iter().eq(plaintexts.keys()) {
-            return None; // Not all accepted contributions are decrypted yet.
+        let mut plaintexts = Vec::new();
+        // Collect accepted plaintexts. Return if some are not decrypted yet.
+        for id in proposer_ids {
+            match self.decryption.get(id) {
+                None | Some(DecryptionState::Ongoing(_)) => return None,
```

Comment:
I'm not sure the returned value of `push` is relevant. You could make this branch return `()` instead and discard the returned value of `push` on the next line.

Reply:
Everything is fine. I misunderstood this line at first.